<?xml version="1.0" encoding="UTF-8" ?>
<testsuites errors="0"
            failures="0"
            tests="13">

  <testsuite name="asserts.game"
    errors="0" failures="0" hostname="" id="" package="asserts" skipped="0"
    tests="13" time="70.72" timestamp="Wed, 20 Dec 2017 12:23:14 GMT">
  
    <testcase name="Check initial game state" classname="asserts.game" time="2.282" assertions="11">

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check multi move for cell" classname="asserts.game" time="5.899" assertions="3">

    

    

    
    </testcase>
  
    <testcase name="Check multi cell reserve" classname="asserts.game" time="2.690" assertions="5">

    

    

    

    

    
    </testcase>
  
    <testcase name="Check horizontal win" classname="asserts.game" time="3.139" assertions="12">

    

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check vertical win" classname="asserts.game" time="6.679" assertions="12">

    

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check right diagonal win" classname="asserts.game" time="6.811" assertions="12">

    

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check left diagonal win" classname="asserts.game" time="2.849" assertions="12">

    

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check horizontal boundery condition" classname="asserts.game" time="6.195" assertions="9">

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check right diagonal boundery condition" classname="asserts.game" time="2.663" assertions="9">

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check left diagonal boundery condition" classname="asserts.game" time="6.639" assertions="9">

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check draw" classname="asserts.game" time="13.14" assertions="105">

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check surrend by owner" classname="asserts.game" time="5.919" assertions="7">

    

    

    

    

    

    

    
    </testcase>
  
    <testcase name="Check surrend by player" classname="asserts.game" time="5.818" assertions="7">

    

    

    

    

    

    

    
    </testcase>
  

  
    <system-err>
        Error while running [Commands / Create Own Game / command]:

  TypeError: Cannot read property &#39;gameList&#39; of undefined
    at Object.createOwnGame [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/createOwnGame.js:12:28)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
  Error while running [Commands / Make Bot Move / command]:

  TypeError: this.perform is not a function
    at Object.makeBotMove [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/makeBotMove.js:11:8)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
  Error while running [Commands / Play / command]:

  TypeError: Cannot read property &#39;game&#39; of undefined
    at Object.play [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/play.js:12:28)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
  Error while running [Commands / Subscribe On Game / command]:

  TypeError: Cannot read property &#39;gameList&#39; of undefined
    at Object.subscribeOnGame [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/subscribeOnGame.js:12:28)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
  Error while running [Commands / Surrend Bot / command]:

  TypeError: this.perform is not a function
    at Object.surrend [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/surrendBot.js:11:8)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
  Error while running [Commands / Throw Exception / command]:

  TypeError: this.perform is not a function
    at Object.throwException [as command] (/Users/js_monk/Projects/tic-tac-toe-10x10/tests/commands/throwException.js:11:8)
    at Module.call (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/module.js:62:34)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/nightwatch/lib/runner/testcase.js:70:29
    at _fulfilled (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:796:13)
    at /Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:556:49
    at runSingle (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:137:13)
    at flush (/Users/js_monk/Projects/tic-tac-toe-10x10/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    </system-err>
  

  
  </testsuite>
</testsuites>
